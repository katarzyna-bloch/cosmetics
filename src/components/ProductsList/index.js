import styled from 'styled-components'
import PropTypes from 'prop-types'
import React from 'react'

import ProductListItem from '../ProductListItem'

const List = styled.div`
  display: flex;
  flex-wrap: wrap;
`

const ProductsList = ({ products }) => {
  return (
    <List>
      {products.map((product) => (
        <ProductListItem 
        {...product} 
        key={product.id}
        />
      ))}
    </List>
  )
}

ProductsList.propTypes = {
  products: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    img: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired,
    price: PropTypes.string.isRequired,
    oldPrice: PropTypes.string,
    hasDiscount: PropTypes.bool,
  })).isRequired,
}

export default ProductsList
