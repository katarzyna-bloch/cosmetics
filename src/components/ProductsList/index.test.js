import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ProductsList from '.'

it('should render correctly', () => {
  const props = {
    products: [
      { name: 'Test', 
        img: 'img.jpg', 
        id: '1', 
        price: '50z≈Ç',
      },
    ],
  }
  const wrapped = shallow(<ProductsList {...props} />)
  expect(toJson(wrapped)).toMatchSnapshot()
})
