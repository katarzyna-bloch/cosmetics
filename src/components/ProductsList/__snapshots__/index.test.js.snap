// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<styled.div>
  <ProductListItem
    id="1"
    img="img.jpg"
    key="1"
    name="Test"
    price="50zÅ‚"
  />
</styled.div>
`;
